hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/DefaultWriteBehindProcessor.java

```diff
@@ -243,14 +244,22 @@ private Map convertToObject(Map<Object, DelayedEntry> batchMap) {
             public boolean run() throws Exception {
                 callBeforeStoreListeners(batchMap.values());
                 final Map map = convertToObject(batchMap);
-                final boolean result = operationType.processBatch(map, mapStore);
+                boolean result;
+                try {
+                    result = operationType.processBatch(map, mapStore);
+                } catch (Exception ex) {
+                    Iterator<Object> keys = batchMap.keySet().iterator();
+                    while (keys.hasNext()) {
```

Comment:
This change might be risky for a maintenance release if somehow a user clears the batch `map` (or removes entries unintentionally) that's passed to `MapStore.storeAll()/deleteAll()`. At the moment in javadoc, we don't mention anything about modification of batch. For instance, I cannot think a reason why someone implements it this way but for following `storeAll()` implementation;

```java
public void storeAll(Map<K, V> map) {
   try {
      [...]
   } catch (Exception e) {
      [...]       
   } finally {
      map.clear();
   }
}
```

@jerrinot: what do you think?